dags/sql/select-find_gaps.sql

```diff
@@ -0,0 +1,77 @@
+--a generic find_gaps function to find gaps in sensor data
```

Comment:
I think this is a great base for a general find_gaps functions! I have some general thoughts:
1)  Can we make this a function? I think this will be really helpful created as a function in postgres, I can think of a couple of use cases, e.g. checking the number of 1 day gap for an individual bluetooth route with in a week, checking a set of link_dir's gaps within peak hours, etc. 
2)  Adding an end_datetime param instead of using ds + 1 day interval. I think this add flexibility to the usage and is a relatively small change
3)  Add a [array] param for selecting sensor_id. default can be all, but sometimes when debugging and investigating gaps for sensors we might be interested in one specific sensor.

I think for point 2 and 3, we could make it into another issue if we want to keep this query as simple as possible, open for discussion

Reply:
Thanks! I'll implement at least 1+2 in this PR, maybe 3. 
I'm also considering a param for returning the list of outages as opposed to the summary if we think this could have wider use.

Reply:
🥳  You can test the new functions in my schema with:

```sql
SELECT * FROM gwolofs.generic_find_gaps(
    '2023-11-20'::date,
    '2023-11-20'::date,
    'intersection_uid'::text,
    'datetime_bin'::text,
    'miovision_api'::text,
    'volumes'::text,
    '10 minutes'::interval,
    '1 minute'::interval
)

SELECT * FROM gwolofs.summarize_gaps_data_check(
    '2023-11-20'::date,
    '2023-11-20'::date,
    'intersection_uid'::text,
    'datetime_bin'::text,
    'miovision_api'::text,
    'volumes'::text,
    '10 minutes'::interval,
    '1 minute'::interval
)
```

A dbadmin will have to create the functions in public schema.

Reply:
splitting them into two function is so helpful! generic_find_gaps is definitely gonna be useful during investigation

Reply:
just tested it with bluetooth, returns an error with data type cause analysis_id is a big int and it was expecting the column to be an integer. We can probs just cast the column to integer explicitly 

```
SELECT * FROM gwolofs.generic_find_gaps(
    '2023-11-20'::date,
    '2023-11-20'::date,
    'analysis_id'::text,
    'datetime_bin'::text,
    'bluetooth'::text,
    'aggr_5min'::text,
    '10 minutes'::interval,
    '1 minute'::interval
)

```

Reply:
actuallllllllllllllly mebbe a text will be better, using it on `link_dir` doesnt work cause its a string

Reply:
I wonder if its possible to do dynamic column type return

Reply:
seems possible! 
https://www.postgresql.org/docs/current/datatype-pseudo.html

Reply:
good call! 
works now with an extra parameter for datatype (null::datatype) 
'''sql
SELECT * FROM gwolofs.generic_find_gaps(
    '2023-11-20'::date,
    '2023-11-20'::date,
    'analysis_id'::text,
    'datetime_bin'::text,
    'bluetooth'::text,
    'aggr_5min'::text,
    '20 minutes'::interval,
    '1 minute'::interval,
    null::bigint
)
'''